includes/site-health/audit-autoloaded-options/helper.php

```diff
@@ -144,13 +144,70 @@ function perflab_aao_get_autoloaded_options_table() {
 	$autoload_summary = perflab_aao_query_autoloaded_options();
 
 	$html_table = sprintf(
-		'<table class="widefat striped"><thead><tr><th scope="col">%s</th><th scope="col">%s</th></tr></thead><tbody>',
-		__( 'Option Name', 'performance-lab' ),
-		__( 'Size', 'performance-lab' )
+		'<table class="widefat striped"><thead><tr><th scope="col">%s</th><th scope="col">%s</th><th scope="col">%s</th></tr></thead><tbody>',
+		esc_html__( 'Option Name', 'performance-lab' ),
+		esc_html__( 'Size', 'performance-lab' ),
+		esc_html__( 'Action', 'performance-lab' )
 	);
 
+	$nonce = wp_create_nonce( 'perflab_aao_update_autoload' );
 	foreach ( $autoload_summary as $value ) {
-		$html_table .= sprintf( '<tr><td>%s</td><td>%s</td></tr>', $value->option_name, size_format( $value->option_value_length, 2 ) );
+		$url            = esc_url_raw(
+			add_query_arg(
+				array(
+					'action'       => 'perflab_aao_update_autoload',
+					'_wpnonce'     => $nonce,
+					'option_name'  => $value->option_name,
+					'autoload'     => 'false',
+					'value_length' => esc_attr( $value->option_value_length ),
+				),
+				admin_url( 'site-health.php' )
+			)
+		);
+		$disable_button = sprintf( '<a class="button" href="%s">%s</a>', esc_url( $url ), esc_html__( 'Disable Autoload', 'performance-lab' ) );
+		$html_table    .= sprintf( '<tr><td>%s</td><td>%s</td><td>%s</td></tr>', esc_html( $value->option_name ), size_format( $value->option_value_length, 2 ), $disable_button );
+	}
+	$html_table .= '</tbody></table>';
+
+	return $html_table;
+}
+
+/**
+ * Gets disabled autoload options table.
+ *
+ * @since n.e.x.t
+ *
+ * @return string HTML formatted table.
+ */
+function perflab_aao_get_disabled_autoloaded_options_table() {
+	$perflab_aao_disabled_options = get_option( 'perflab_aao_modified_options', array() );
+
+	if ( empty( $perflab_aao_disabled_options ) ) {
+		return '';
+	}
+
```

Comment:
Instead of storing the option size from the time of disabling autoload for the option, here you could run a database query that gets those sizes.

It could look rather similar to the other query, however you would need to use something like `WHERE option_name IN (...)` with the concrete list of the options for which autoload was disabled. This way the table will always display up to date values.

It also brings a real user benefit: Maybe an option is very large, but later the plugin developer modifies what is stored there, so the option becomes smaller after an update. This table will then show it and the admin may consider reverting it back to autoload.

Reply:
Thank you for bringing this to my attention. Initially, I considered using a query, but to avoid it, I stored the option value. However, I overlooked updating the option even if it is disabled. In https://github.com/WordPress/performance/pull/1048/commits/4e857a448d883663e79f3b48e53376d97fb1269f, I have made the necessary updates as per your direction.